### PR TITLE
Restructure Terraform code to use existing modules and access requirements

### PR DESCRIPTION
**Pull request by MartyFox**
### Jira link (if applicable)
https://tools.hmcts.net/jira/browse/DTSPO-17828

### Change description ###
Renaming terraform component to be more relevant, updating code to create KV and MI via existing module, adding group access to MI for read only

### Checklist

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

File: components/versionreporting/02-variabes.tf [hunk 1]
```diff
@@ -45,7 +37,6 @@ variable "containers_partitions" {
     helmcharts   = "/namespace"
     renovate     = "/repository"
     docsoutdated = "/docTitle"
-    netflow      = "/netflow"
```
Comment (reespozzi) [inline components/versionreporting/02-variabes.tf:48]: Assuming this is not needed anymore and not accidental?

File: components/versionreporting/02-variabes.tf [hunk 2]
```diff
@@ -6,18 +6,10 @@ variable "environment" {
   default = "stg"
```
Comment (reespozzi) [inline components/versionreporting/02-variabes.tf:6]:
```suggestion
  default = "ptl"
```
?

File: components/versionreporting/02-variabes.tf [hunk 3]
```diff
@@ -6,18 +6,10 @@ variable "environment" {
...
 }
 
-variable "product" {
-  default = "sds-platform"
-}
-
```
Comment (reespozzi) [inline components/versionreporting/02-variabes.tf:12]: Can we keep product name? May cause tagging errors / confusion

Comment (reespozzi) [reply components/versionreporting/02-variabes.tf:12]: I see var.serviceName in common tags but struggling to see where defined?

**Review by JoshuaGlassKainos — APPROVED** [review]

File: components/versionreporting/02-variabes.tf (hunk 1 above)
Comment (MartyFox) [reply components/versionreporting/02-variabes.tf:48]: I believe Felix had it as a poc for something but theres no data

File: components/versionreporting/02-variabes.tf (hunk 2 above)
Comment (MartyFox) [reply components/versionreporting/02-variabes.tf]: moving to CFT PTL now

File: components/versionreporting/02-variabes.tf (hunk 3 above)
Comment (MartyFox) [reply components/versionreporting/02-variabes.tf]: resolved

**Pull request merged**